listenbrainz/webserver/login/provider.py

```diff
@@ -33,16 +36,23 @@ def get_user():
     data = s.get('oauth2/userinfo').json()
     musicbrainz_id = data.get('sub')
     musicbrainz_row_id = data.get('metabrainz_user_id')
-    user_exists_check = db_user.get_by_mb_id(musicbrainz_id)
-    user = db_user.get_or_create(musicbrainz_row_id, musicbrainz_id)
-    if user:
-        if user_exists_check is None:
-            ts.set_empty_cache_values_for_user(musicbrainz_id)
-        if not user['musicbrainz_row_id']:
-            db_user.update_musicbrainz_row_id(musicbrainz_id, data['metabrainz_user_id'])
-        return User.from_dbrow(user)
+
+    user = db_user.get_by_mb_row_id(musicbrainz_row_id, musicbrainz_id)
+    user_email = None
+    if mb_engine:
+        user_email = mb_editor.get_editor_by_id(musicbrainz_row_id)['email']
```

Comment:
we should also check that this is a _valid_ email address (column `email_confirm_date`) in the musicbrainz editor row

Reply:
after looking at the musicbrainz server email validation process, this might not be necessary. There appears to never be a situation where an email is set but email_confirm_date is null.

Reply:
email is null and email_confirm_date is non-null on a public data dump. I'm not sure this is a usecase that we need to support

Reply:
We don't use email_confirm_date in LB and do not store, so I don't think we need it in our dumps anyways.

Reply:
I meant to say here that LB _should_ check email confirm date, in order to know that the email address that we're using is a valid email.
However, it turns out that musicbrainz always sets these in parallel. If the email is null then  email_confirm_date, and if email is set, there's always a email_confirm_date. The only situation where this doesn't happen is on a public data dump, but I don't think we need to support this case. We can probably ignore this comment.